README.md

```diff
@@ -45,11 +41,36 @@ require(['userlike-messenger'], function (userlike) {
 
 ------
 
-## Examples
+## Usage
 
-### Create the API
+Any function from Userlike Messenger API returns an Action Result that you have to handle based on your needs (e.g. throw an error, ignore the result).
+The Action result is a discriminated (or tagged) Union Type that either represents a `success` or `error`.
+
+For example, when creating the messenger you will either receive an Action Result of
```

Comment:
I think this section doesn't describe usage at all but just talks about error handling, so instead I'd place it after `Examples` and call it `Error handling` and I'd keep the existing `Usage`.

Do you think if it would be better to word it something like follows?

> ## Error handling
>Messenger API never throws an error. Instead it returns an [`ActionResult`](https://github.com/userlike/messenger/blob/master/packages/messenger-internal/src/ActionResult.ts#L4) which represents either a successful outcome or an erroneous outcome. In technical terms, `ActionResult` is a [tagged union](https://en.wikipedia.org/wiki/Tagged_union) of `ActionSuccess` and `ActionError`, similar to [Rust's `Result<E, T>`](https://doc.rust-lang.org/std/result/).

I think the javascript object snippets are somewhat confusing due to `value: messengerApi` bit. It would be just easier to put a full example that creates a messenger api and handles errors (or just link to the existing example).